npm/test/verifycheckscompletedanddisplaybanner.test.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  displayChecksCompletedBanner,
+  haveFileChecksCompleted
+} from "../src/filechecks/verify-checks-completed-and-display-banner"
+
+test("haveFileChecksCompleted returns true if file checks have completed", () => {
+  const mockFileChecksResponse = {
+    antivirusProcessed: 2,
+    checksumProcessed: 2,
+    ffidProcessed: 2,
+    totalFiles: 2
+  }
+
+  const checksCompleted: boolean = haveFileChecksCompleted(
+    mockFileChecksResponse
+  )
+  expect(checksCompleted).toBe(true)
+})
+
+test("haveFileChecksCompleted returns false if file checks have not completed", () => {
+  const mockFileChecksResponse = {
+    antivirusProcessed: 1,
+    checksumProcessed: 2,
+    ffidProcessed: 1,
+    totalFiles: 2
+  }
+  const checksCompleted: boolean = haveFileChecksCompleted(
+    mockFileChecksResponse
+  )
+
+  expect(checksCompleted).toBe(false)
+})
+
+test("haveFileChecksCompleted returns false if null was passed in", () => {
+  const mockFileChecksResponse = null
+  const checksCompleted: boolean = haveFileChecksCompleted(
+    mockFileChecksResponse
+  )
+
+  expect(checksCompleted).toBe(false)
+})
+
+test("displayChecksCompletedBanner unhides the button and displays banner when called", () => {
+  document.body.innerHTML = `<div id="file-checks-completed-banner" hidden></div>
+                            <a id="file-checks-continue" class="govuk-button--disabled" disabled></a>`
+  displayChecksCompletedBanner()
+
+  const notificationBanner = document.querySelector(
+    "#file-checks-completed-banner"
+  )
+  const continueButton = document.querySelector("#file-checks-continue")
+
+  if (notificationBanner && continueButton) {
+    expect(notificationBanner.getAttribute("hidden")).toBeNull()
+    expect(
+      continueButton.classList.contains("govuk-button--disabled")
+    ).toBeFalsy()
+    expect(continueButton.hasAttribute("disabled")).toBeFalsy()
+  }
+
+})
+
+test("displayChecksCompletedBanner displays banner but can't unhide the button", () => {
+  document.body.innerHTML = `<div id="file-checks-completed-banner" hidden></div>`  // no button exists in the HTML
+  displayChecksCompletedBanner()
+
+  const notificationBanner = document.querySelector(
+    "#file-checks-completed-banner"
+  )
+  const continueButton = document.querySelector("#file-checks-continue")
+  expect(continueButton).toBeNull()
+  expect(notificationBanner).not.toBeNull()
+  //Keep typescript happy
+  if (notificationBanner) {
```

Comment:
It's good to avoid `if` statements in tests. If someone breaks the HTML fixture on line 64, this test would still pass because the `if` evaluates to `true`.

Better to do something that will result in an a failed test or an error, so that we spot the problem. I not familiar enough with TypeScript to know what the idiomatic thing to do is, but can you use the exclamation mark syntax instead?

```
expect(notificationBanner!.getAttribute("hidden")).toBeNull()
```